collect_app/src/main/java/org/odk/collect/android/location/client/GoogleFusedLocationClient.java

```diff
@@ -120,15 +120,15 @@ public void stop() {
 
     @SuppressLint("MissingPermission") // Permission checks for location services handled in widgets
     public void requestLocationUpdates(@NonNull LocationListener locationListener) {
-        if (!isMonitoringLocation()) {
+        if (!isMonitoringLocation() && googleApiClient.isConnected()) {
             fusedLocationProviderApi.requestLocationUpdates(googleApiClient, createLocationRequest(), this);
         }
 
         this.locationListener = locationListener;
     }
 
     public void stopLocationUpdates() {
-        if (!isMonitoringLocation()) {
+        if (!isMonitoringLocation() || !googleApiClient.isConnected()) {
```

Comment:
I think it's possible for a `locationListener` to be set and `googleApiClient` not to be connected. In that case, `locationListener` would not be set to `null`. I think that's not a good state to be in.

Reply:
Ah you mean that locationListener won't be cleared in such a case?

Reply:
Whoops, missed this. Yes, exactly. That means `isMonitoringLocation()` would evaluate to `true`. I don't know if we can really reach that state from Collect but I'd rather not risk it.

Reply:
Fixed.